cli/command/stack/kubernetes/convert.go

```diff
@@ -8,13 +8,58 @@ import (
 	"strings"
 
 	"github.com/docker/cli/cli/compose/loader"
-	composeTypes "github.com/docker/cli/cli/compose/types"
+	"github.com/docker/cli/cli/compose/schema"
 	composetypes "github.com/docker/cli/cli/compose/types"
 	"github.com/docker/cli/kubernetes/compose/v1beta1"
 	"github.com/docker/cli/kubernetes/compose/v1beta2"
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// FromCompose converts types.Config (compose) to the specified stack version.
+// It errors out if the version is not supported.
+func FromCompose(version string, stderr io.Writer, name string, cfg *composetypes.Config) (Stack, error) {
+	switch version {
+	case "v1beta1":
+		return v1beta1FromCompose(stderr, name, cfg)
+	case "v1beta2":
+		return v1beta2FromCompose(stderr, name, cfg)
+	default:
+		return Stack{}, errors.Errorf("stack version %s unsupported", version)
+	}
+}
+
+func v1beta1FromCompose(stderr io.Writer, name string, cfg *composetypes.Config) (Stack, error) {
+	cfg.Version = v1beta1.MaxComposeVersion
+	st, err := v1beta2FromCompose(stderr, name, cfg)
+	if err != nil {
+		return Stack{}, err
+	}
+	res, err := yaml.Marshal(cfg)
+	if err != nil {
+		return Stack{}, err
+	}
+	// reload the result to check that it produced a valid 3.5 compose file
+	resparsedConfig, err := loader.ParseYAML(res)
+	if err != nil {
+		return Stack{}, err
+	}
+	if err = schema.Validate(resparsedConfig, v1beta1.MaxComposeVersion); err != nil {
+		return Stack{}, errors.Wrapf(err, "the compose yaml file is invalid with v%s", v1beta1.MaxComposeVersion)
+	}
+
+	st.ComposeFile = string(res)
```

Comment:
wondering if there's consequences of not passing the original file content; is it to normalize the YAML? (if so, we may need some comments explaining)